src/transformStateWithClones.js

```diff
@@ -8,6 +8,33 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateHistory = [];
+  let previousState = { ...state };
+
+  for (const action of actions) {
+    const { type } = action;
+
+    let newState = { ...previousState };
+
+    if (type === 'addProperties') {
+      newState = Object.assign({}, previousState, action.extraData);
+    }
+
+    if (type === 'removeProperties') {
+      for (const key of action.keysToRemove) {
+        delete newState[key];
+      }
+    }
+
+    if (type === 'clear') {
+      newState = {};
+    }
```

Comment:
[CODE STYLE]: Use a switch statement instead of multiple if statements when you have a limited amount of conditions that are based on the same expression (`type` in this case). This will make your code more readable and organized.